bring up-to-date with canonical-kubernetes

Cynerva opened the pull request:
Work for https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/70

This pulls in additional functionality from canonical-kubernetes (flannel, ingress, etc) while trying to maintain a low machine footprint.

This includes an updated README as well, [see here](https://github.com/juju-solutions/bundle-kubernetes-core/tree/gkk/converged).

@chuckbutler @mbruzek @wwwtyro @marcoceppi

Cynerva committed:
bring up-to-date with canonical-kubernetes

Cynerva committed:
update README

Started from the Canonical Distribution of Kubernetes README,
tweaked from there

wwwtyro commented:
:+1: LGTM

lazypower commented:
This looks fine to me. _disclaimer_ I haven't combed the README for consistency and accuracy, but the bundle itself looks good.

lazypower commented:
@Cynerva - i know this is late in the game, but we did just rev the k8s charms to -4

would you mind bumping the charm revisions in here to reflect that?

mbruzek commented on README.md:
```diff
@@ -1,123 +1,404 @@
 
 Etcd is used as a key-value store for the Kubernetes cluster. For reliability
-the cluster defaults to three instances in this bundle.
+the bundle defaults to three instances in this cluster.
```
The number of etcd units are 1 in this new bundle. We should change 'three' to 'one' here.

mbruzek commented on bundle.yaml:
```diff
@@ -1,21 +1,85 @@
+  1:
+  2:
+  3:
+applications:
```
The key 'applications' is a v4 bundle convenation that is not yet widely supported. @chuckbutler had to change this key back to 'services' in the other bundle for it to work with our testing tools.

mbruzek commented on bundle.yaml:
```diff
@@ -1,21 +1,85 @@
+  3:
+applications:
+  "kubernetes-master":
+    charm: "cs:~containers/kubernetes-master-3"
```
We **just** incremented the version of the kubernetes-master charm last night. s/3/4/

mbruzek commented on bundle.yaml:
```diff
@@ -1,21 +1,85 @@
+      "gui-x": "450"
+      "gui-y": "550"
+  "kubernetes-worker":
+    charm: "cs:~containers/kubernetes-worker-3"
```
We also published a new kubernetes-worker with fixes. Can you change this to 4 as well?

mbruzek reviewed (commented):
Great work on this README and bundle! I have a few changes that would make this more accurate.

Cynerva committed:
update bundle with latest charm revs, use services instead of applications

Cynerva committed:
update README with correct etcd unit count

Cynerva commented:
Updated:
- Bumped revisions for kubernetes-worker, kubernetes-master, and etcd
- Used `services:` instead of `applications:`
- Fixed etcd unit count in README

Have not tested with the bundle changes yet; will do so later today.

Cynerva committed:
Move etcd to machine 0 with kubeapi-load-balancer

Cynerva commented:
Updated:
- Moved etcd to machine 0 (shared with kubeapi-load-balancer)

Again, not tested yet.

mbruzek commented:
Thanks for addressing my comments @Cynerva ! +1 to these changes.
